### PR TITLE
Improve graphics code style and documentation (adds Checkstyle)

### PR DESCRIPTION
**Pull request by michaelzangl**
This adds:
- Checkstyle checks for graphics+common package.
- Copyright header as default for all files with checkstyle check to enforce it
- Some files are changed to unix line endings (checkstyle check for it)
- Unified some constant names (values->VALUES)
- Extracted some constants.
- Unify image method order
- Split drawMapObject in multiple methods
- Add a lot of Javadoc
- Add some missing copyright comments.

### CODE REVIEW

File: jsettlers.graphics/src/jsettlers/graphics/image/ImageIndexImage.java [hunk 1]
```diff
@@ -31,6 +38,30 @@
 	private final float umax;
 	private final float vmax;
 
+	private static final float[] TEMP_BUFFER = new float[5 * 6];
```
Comment (andreas-eberle) [inline jsettlers.graphics/src/jsettlers/graphics/image/ImageIndexImage.java]: @michaelzangl: Didn't we agree on naming such "not really constant" variables in normal camel case?

File: jsettlers.graphics/src/jsettlers/graphics/image/SingleImage.java [hunk 2]
```diff
@@ -148,7 +152,7 @@ public TextureHandle getTextureIndex(GLDrawContext gl) {
 		return this.texture;
 	}
 
-	static private float[] tmpBuffer = new float[] {
+	private static float[] TMP_BUFFER = new float[] {
```
Comment (andreas-eberle) [inline jsettlers.graphics/src/jsettlers/graphics/image/SingleImage.java]: Again, this variable isn't really constant.

**Comment by andreas-eberle** [comment]
The material priority panel is broken. @michaelzangl: Can you have a look?

unbenannt

**Comment by michaelzangl** [comment]
Material panel is fixed.

**Pull request merged**